core/schema/sdk.go

```diff
@@ -99,25 +99,42 @@ func (s *moduleSchema) sdkForModule(
 	return s.newModuleSDK(ctx, query, sdkMod, dagql.Instance[*core.Directory]{})
 }
 
-// parse and return the name and suffix from sdkName
-// e.g.
+// parse and validate the name and version from sdkName
 //
 // for sdkName with format <sdk-name>@<version>, it returns
-// '<sdk-name>' as name and '@<version>' as suffix AND
+// '<sdk-name>' as name and '@<version>' as suffix.
 //
-// for sdkName with format <sdk-name>, it returns <sdk-name> with
-// empty suffix.
-func parseSDKName(sdkName string) (string, string) {
+// If sdk is one of go/python/typescript and <version>
+// is specified, we return an error as those sdk don't support
+// specific version
+//
+// if sdk is one of php/elixir and version is not specified,
+// we defaults the version to [engine.Tag]
+func parseAndValidateBuiltinSDKName(sdkName string) (string, string, error) {
```

Comment:
We can keep this called `parseSDKName` - I'm not a fan of overly verbose names, they make the code much harder to read :cry: 

One other suggestion that might improve readability in this area if you feel like it - we could make `SDKGo` and similar into an enum with a new `type SDK string`.